### PR TITLE
Reservation Endpoints

### PR DESCRIPTION
**Pull request by piero-vic**
# Reservation Endpoints

- Added the index and show actions for the reservation controller.
- Users get an error message if the record they are trying to access is not found. The response looks like this.
```json
{
    "error": "Reservation not found"
}
```
## Notes:
- We set up the controllers without a namespace. Let us know if this is okay or if we need to change it. 
- We are also a little bit unsure if we should add the create and delete actions before implementing proper authorization. I hope to hear from you guys.

### CODE REVIEW

**Review by Reem-lab — CHANGES_REQUESTED** [review]
Hi @piero-vic  and @German-Cobian,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
but you still have some changes to do so check it below_

## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯
  
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def show
+    reservation = Reservation.find_by(id: params[:id])
+    if reservation.nil?
+      render status: 404, json: { error: 'Reservation not found' }.to_json
+    else
+      render json: reservation.to_json
+    end
+  end
+end
```
[the remark this answers is not in the document]
Comment (piero-vic) [reply app/controllers/reservations_controller.rb:14]: Cool @Reem-lab. I just had concerns about these actions. If you think we should implement them now we'll do it.

**Comment by piero-vic** [comment]
@Reem-lab. You haven't said anything about the namespace. We'll do it similar to the way you did on #21. :+1:

**Comment by Reem-lab** [comment]
Hi @piero-vic 
 - I think it will work like that but ok it is better to do it with namespace because it is like the material. 🚀🔥💪🏻

**Review by Reem-lab — APPROVED** [review]
## Hi @piero-vic  and @German-Cobian  👋🏻👋🏻

# STATUS: APPROVED 🥳 🥳 🥳 



#### Your project is great. 🎉 I approved it. There is nothing else to say other than... it's time to merge it.

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**